### PR TITLE
CB-14307 Extend FreeIPA, DataLake and DataHub interfaces with change image catalog.

### PR DESCRIPTION
**Pull request by pkedvessy**
Adds dummy implementations so we can work in parallel on the real implementation and on the related CLI changes.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1Controller.java [hunk 1]
```diff
@@ -587,6 +588,12 @@ public DistroXSyncCmV1Response syncComponentVersionsFromCmByCrn(@ResourceCrn Str
         return launchSyncComponentVersionsFromCm(NameOrCrn.ofCrn(crn));
     }
 
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.POWERUSER_ONLY)
+    public void changeImageCatalog(@ResourceName @NotEmpty String name, @Valid @NotNull ChangeImageCatalogV4Request changeImageCatalogRequest) {
+        // FIXME: Implement this!
```
Comment (pkedvessy) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1Controller.java:594]: API first approach so we can work in parallel on the CLI.
It is going to be implemented as a part of CB-14308

File: datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java [hunk 2]
```diff
@@ -343,6 +346,12 @@ public FlowIdentifier rotateAutoTlsCertificatesByCrn(@ResourceCrn @TenantAwarePa
         return certRotationService.rotateAutoTlsCertificates(sdxCluster, rotateCertificateRequest);
     }
 
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.POWERUSER_ONLY)
+    public void changeImageCatalog(@ResourceName @NotEmpty String name, @Valid @NotNull  SdxChangeImageCatalogRequest changeImageCatalogRequest) {
+        // FIXME: Implement this!
```
Comment (pkedvessy) [inline datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java:352]: API first approach so we can work in parallel on the CLI.
It is going to be implemented as a part of CB-14309

File: freeipa/src/main/java/com/sequenceiq/freeipa/controller/FreeIpaV1Controller.java [hunk 3]
```diff
@@ -308,4 +309,10 @@ public FlowIdentifier retry(@ResourceCrn @NotEmpty String environmentCrn) {
         String accountId = crnService.getCurrentAccountId();
         return retryService.getRetryableFlows(environmentCrn, accountId);
     }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = EDIT_ENVIRONMENT)
+    public void changeImageCatalog(@ResourceCrn @NotEmpty String environmentCrn, @Valid @NotNull ChangeImageCatalogRequest changeImageCatalogRequest) {
+        // FIXME: Implement this!
```
Comment (pkedvessy) [inline freeipa/src/main/java/com/sequenceiq/freeipa/controller/FreeIpaV1Controller.java:316]: API first approach so we can work in parallel on the CLI.
It is going to be implemented as a part of CB-14310

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1Controller.java (hunk 1 above)
Comment (Bajzathd) [reply core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXV1Controller.java:594]: I think this comment would be better in the code, not just the PR

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/ChangeImageCatalogV4Request.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+public class ChangeImageCatalogV4Request {
+
+    @Size(max = 100, min = 5, message = "The length of the image catalog has to be in range of 5 to 100")
+    @Pattern(regexp = "(^[a-z][-a-z0-9]*[a-z0-9]$)",
+            message = "Image catalog can only contain lowercase alphanumeric characters and hyphens and has start with an alphanumeric character")
```
Comment (Bajzathd) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/ChangeImageCatalogV4Request.java:17]: I think you have missed a to:

> has to start with

**Review by Bajzathd — APPROVED** [review]

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/ChangeImageCatalogV4Request.java (hunk 4 above)
Comment (pkedvessy) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/ChangeImageCatalogV4Request.java:17]: Donno, copy paste from an other operation :D But seems like yes

**Review by TheTinkerDad — APPROVED** [review]

**Comment by pkedvessy** [comment]
retest this please

**Pull request merged**